src/useGetModifiers.tsx

```diff
@@ -41,10 +56,19 @@ export function useGetModifiers(
 
     const isOutside = Boolean(displayMonth && !isSameMonth(date, displayMonth));
 
-    const isDisabled = Boolean(
-      disabled && dateMatchModifiers(date, disabled, dateLib)
+    const isBeforeStartMonth = Boolean(
+      startMonth && isBefore(date, startOfMonth(startMonth))
+    );
+
+    const isAfterEndMonth = Boolean(
+      endMonth && isAfter(date, endOfMonth(endMonth))
     );
 
+    const isDisabled =
```

Comment:
Why disabling these days, instead of actually _hiding_ them ? (In the line below, we calculate the `isHidden` for that.)

Reply:
For me it makes sense to hide, I left this implementation in the draft because I first started with it, and then understanding better the behavior of the calendar I discovered that we have different behaviors for days before the `startMonth` month and the ones after the `endMonth`, so I wanted to check with you first what would the preferred approach if we should commit to the same behavior for both sides of the `startMonth` and `endMonth` range.

With your feedback I have enough info to finish this PR 👍 
I'll update the logic and add some tests to mark the PR as ready.